lib/chef/provider/user/solaris.rb

```diff
@@ -46,21 +46,14 @@ def manage_user
         end
 
         def check_lock
-          shadow_line = shell_out!("getent", "shadow", new_resource.username).stdout.strip rescue nil
+          user = IO.read(@password_file).match(/^#{Regexp.escape(@new_resource.username)}:(\*?\w+\*?)/)
```

Comment:
If you made this `/^#{Regexp.escape(@new_resource.username)}:([^:]*):/` it would be slightly more robust since you are checking the hash bit in a second regexp below and just want to extract the field I think.